### ISSUE TITLE
Home Accessories updating slowly 

### ISSUE THREAD

**Issue by Gio76** [hassio-addons/addon-homebridge#71]
# Problem/Motivation

When launching the **Home App**, all Home Assistant accessories _(added via Homedridge addon)_ are taking **several seconds to update their status** _(up to 10 most times)_, whilst the ones added natively like **Eve Elgato** or **Philips Hue** lights are readily available for interaction. This is becoming very frustrating to deal with, especially considering that there are instances where things are running fast and smooth but more times than not, I am stuck looking at my iPhone screen waiting for the device to be ready so I can turn it on/off. 

I am not sure if this is a known issue or it's simply expected behaviour. I did some research online and couldn't fine any conclusive answer/solution to my problem hence this Issue post.  

_**My Hardware setup:**_
- Raspberry Pi 3 Model B
- SanDisk 32GB Extreme PRO Micro SD 667x R100/W90
- Pi powered by a 5V 2.5A power supply
- Internet 200 Mbps Fibre Optic connection
- Google Wifi Mesh network _(Pi connected to the Primary Point via cat7 ethernet cable)_
- iPhone X

**_My Software setup:_**
- Hassio 0.60
- Supervisor 0.79
- Host OS 1.1
- Homebridge addon 2.2.0
- iOS 11.2.1

_**Homebridge addon Options:**_
```json
{
  "log_level": "info",
  "avahi_interfaces": "",
  "avahi_hostname": "",
  "avahi_domainname": "local",
  "enable_ipv6": false,
  "packages": [],
  "init_commands": [],
  "plugins": []
}
```

**_Homebridge config.json:_**
```json
{
  "bridge": {
    "name": "Home Assistant",
    "username": "B8:27:EB:A8:84:FA",
    "port": 51826,
    "pin": "087-70-688"
  },
  "description": "Homebridge for Home Assistant",
  "accessories": [],
  "platforms": [
    {
      "platform": "HomeAssistant",
      "name": "HomeAssistant",
      "host": "https://MY_HA.duckdns.org:8123",
      "password": "MY_PASSWORD",
      "default_visibility": "visible",
      "supported_types": [
        "automation",
        "binary_sensor",
        "climate",
        "cover",
        "device_tracker",
        "fan",
        "group",
        "light",
        "lock",
        "media_player",
        "remote",
        "scene",
        "script",
        "sensor",
        "switch"
      ],
      "logging": true,
      "verify_ssl": false
    }
  ]
}
```

It's worth noting that I initially had the **"host"** set to `http://hassio/homeassistant`, **password** was empty and **"verify_ssl"** set to `true`, however that was giving me the worse results. I also tested switching around **"enable_ipv6"**, I tried both `true` and `false`, but didn't see any noticeable difference.

## Expected behaviour

Been able to interact with accessories immediately _(or at least within a few seconds)_ after launching the **Home App**.

## Actual behavior

Every time I launch the **Home App**, I need to wait up to 10 seconds for the Home Assistant accessories to be available since they 'stuck' on **Updating**, as per below screenshot _(please note that the kitchen oven is showing correctly since it's NOT a HA device - it's in fact connected to a Bluetooth Eve Elgato smart switch)_:

![img_0181](https://user-images.githubusercontent.com/26906928/34647963-2e2ba42a-f3c3-11e7-987d-8dcc7e50fbf0.JPG)

## Steps to reproduce

- Have Hassio and Homebridge 2.2.0 addon installed
- After correctly setting up Homebridge in iOS, launch Home App
- Look out for any delays in the accessories status

**Comment by frenck** [comment]
Recently resolved and will be fine-tuned in the next release.
**Ensure you've installed v0.77 of Hass.io/Supervisor or newer.**

Next, follow the recommendation from the guide :
> We recommend using `http://hassio/homeassistant` as the `host` with an
> empty `password`, which allows Homebridge to talk to Home Assistant directly.

If you have the latest Supervisor installed (0.77 or newer), Homebridge will create a WebSocket connection with Home Assistant, providing live updates instead of polling for them.

**Comment by Gio76** [comment]
Thanks for the prompt reply @frenck! 

> Ensure you've installed v0.77 of Hass.io/Supervisor or newer.

As mentioned in my previous post, I already have the latest Supervisor installed (v0.79). 

I followed your suggestion and switched the `host` back to `http://hassio/homeassistant`, then updated **Hassio** to **0.60.1**, rebooted and noticed right away some improvements but they lasted only a few minutes...the huge delays started again 😞

**Comment by frenck** [comment]
@Gio76 Could you provide me some debug logs?

**Comment by Gio76** [comment]
There you go @frenck 

```
[s6-init]` making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] 10-homebridge: applying... 
[fix-attrs.d] 10-homebridge: exited 0.
[fix-attrs.d] 20-dbus: applying... 
[fix-attrs.d] 20-dbus: exited 0.
[fix-attrs.d] 30-avahi: applying... 
[fix-attrs.d] 30-avahi: exited 0.
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Hass.io Add-on: Homebridge v2.2.0
 HomeKit support for your Home Assistant instance using Homebridge
 From: Community Hass.io Add-ons
 By: Franck Nijhof <frenck@addons.community>
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
Log level is set to INFO
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] 02-updates.sh: executing... 
INFO: You are running the latest version of this add-on
[cont-init.d] 02-updates.sh: exited 0.
[cont-init.d] 20-dbus.sh: executing... 
[cont-init.d] 20-dbus.sh: exited 0.
[cont-init.d] 30-avahi.sh: executing... 
[cont-init.d] 30-avahi.sh: exited 0.
[cont-init.d] 50-config.sh: executing... 
[cont-init.d] 50-config.sh: exited 0.
[cont-init.d] 60-ipv6-hotfix.sh: executing... 
patching file /usr/lib/node_modules/homebridge/node_modules/hap-nodejs/lib/util/eventedhttp.js
[cont-init.d] 60-ipv6-hotfix.sh: exited 0.
[cont-init.d] 80-user-packages.sh: executing... 
[cont-init.d] 80-user-packages.sh: exited 0.
[cont-init.d] 81-user-scripts.sh: executing... 
[cont-init.d] 81-user-scripts.sh: exited 0.
[cont-init.d] 82-user-plugins.sh: executing... 
[cont-init.d] 82-user-plugins.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
starting version 3.2.4
[services.d] done.
Found user 'avahi' (UID 86) and group 'avahi' (GID 86).
Successfully dropped root privileges.
avahi-daemon 0.6.32 starting up.
WARNING: No NSS support for mDNS detected, consider installing nss-mdns!
Successfully called chroot().
Successfully dropped remaining capabilities.
No service file found in /etc/avahi/services.
*** WARNING: Detected another IPv4 mDNS stack running on this host. This makes mDNS unreliable and is thus not recommended. ***
Joining mDNS multicast group on interface hassio.IPv4 with address 172.30.32.1.
New relevant interface hassio.IPv4 for mDNS.
Joining mDNS multicast group on interface eth0.IPv4 with address 192.168.86.20.
New relevant interface eth0.IPv4 for mDNS.
Network interface enumeration completed.
Registering new address record for 172.30.32.1 on hassio.IPv4.
Registering new address record for 192.168.86.20 on eth0.IPv4.
Server startup complete. Host name is hassio.local. Local service cookie is 2273548978.
*** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node>
*** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node&f=DNSServiceRegister>
[2018-1-8 00:03:05] Loaded plugin: homebridge-homeassistant
[2018-1-8 00:03:06] Registering platform 'homebridge-homeassistant.HomeAssistant'
[2018-1-8 00:03:06] ---
[2018-1-8 00:03:06] Loaded config.json with 0 accessories and 1 platforms.
[2018-1-8 00:03:06] ---
[2018-1-8 00:03:06] Loading 1 platforms...
[2018-1-8 00:03:06] [HomeAssistant] Initializing HomeAssistant platform...
[2018-1-8 00:03:06] [HomeAssistant] Fetching HomeAssistant devices.
[2018-1-8 00:03:06] Loading 0 accessories...
[2018-1-8 00:03:56] [HomeAssistant] Received event: ping
[2018-1-8 00:03:56] [HomeAssistant] Initializing platform accessory 'Office Real Feel'...
[2018-1-8 00:03:56] [HomeAssistant] Initializing platform accessory 'Gate'...
[2018-1-8 00:03:56] [HomeAssistant] Initializing platform accessory 'Office °C'...
[2018-1-8 00:03:56] [HomeAssistant] Initializing platform accessory 'Kitchen Fan'...
[2018-1-8 00:03:56] [HomeAssistant] Initializing platform accessory 'Tower Fan'...
[2018-1-8 00:03:56] [HomeAssistant] Initializing platform accessory 'Ring Ding'...
[2018-1-8 00:03:56] [HomeAssistant] Initializing platform accessory 'Ring Motion'...
[2018-1-8 00:03:56] [HomeAssistant] Initializing platform accessory 'Bedroom °C'...
[2018-1-8 00:03:56] [HomeAssistant] Initializing platform accessory 'Living Room °C'...
[2018-1-8 00:03:56] [HomeAssistant] Initializing platform accessory 'TV Audio'...
[2018-1-8 00:03:56] [HomeAssistant] Initializing platform accessory 'LED Strip'...
[2018-1-8 00:03:56] [HomeAssistant] Initializing platform accessory 'Wall Fan'...
[2018-1-8 00:03:57] [HomeAssistant] Initializing platform accessory 'TV Power'...
[2018-1-8 00:03:57] [HomeAssistant] Initializing platform accessory 'TV Play'...
[2018-1-8 00:03:57] [HomeAssistant] Initializing platform accessory 'Bedroom Air Con'...
[2018-1-8 00:03:57] [HomeAssistant] Initializing platform accessory 'Bedroom Play'...
[2018-1-8 00:03:57] [HomeAssistant] Initializing platform accessory 'Bedroom Fan'...
[2018-1-8 00:03:57] [HomeAssistant] Initializing platform accessory 'Bedroom TV Audio'...
[2018-1-8 00:03:57] [HomeAssistant] Initializing platform accessory 'Bedroom TV'...
[2018-1-8 00:03:57] [HomeAssistant] Initializing platform accessory 'Office Air Con'...
[2018-1-8 00:03:57] [HomeAssistant] Initializing platform accessory 'Office Fan'...
[2018-1-8 00:03:57] [HomeAssistant] Initializing platform accessory '3D Printer Light'...
Setup Payload:
X-HM://0023M5JLSN716
Scan this code with your HomeKit app on your iOS device to pair with Homebridge:
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
                                                      
Or enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ 087-70-688 │     
    └────────────┘     
                       
[2018-1-8 00:03:57] Homebridge is running on port 51826.
[2018-1-8 00:04:00] [HomeAssistant] Received event: {"event_type": "state_changed", "data": {"entity_id": "sensor.ram_used_2", "old_state": {"entity_id": "sensor.ram_used_2", "state": "607.7", "attributes": {"unit_of_measurement": "MiB", "friendly_name": "RAM used", "icon": "mdi:memory", "homebridge_hidden": true, "emulated_hue_hidden": true, "google_assistant": false}, "last_changed": "2018-01-07T17:03:29.953613+00:00", "last_updated": "2018-01-07T17:03:29.953613+00:00"}, "new_state": {"entity_id": "sensor.ram_used_2", "state": "614.7", "attributes": {"unit_of_measurement": "MiB", "friendly_name": "RAM used", "icon": "mdi:memory", "homebridge_hidden": true, "emulated_hue_hidden": true, "google_assistant": false}, "last_changed": "2018-01-07T17:04:00.769858+00:00", "last_updated": "2018-01-07T17:04:00.769858+00:00"}}, "origin": "LOCAL", "time_fired": "2018-01-07T17:04:00.770080+00:00"}
[2018-1-8 00:04:00] [HomeAssistant] Received event: {"event_type": "state_changed", "data": {"entity_id": "sensor.cpu_used", "old_state": {"entity_id": "sensor.cpu_used", "state": "15", "attributes": {"unit_of_measurement": "%", "friendly_name": "CPU used", "icon": "mdi:memory", "homebridge_hidden": true, "emulated_hue_hidden": true, "google_assistant": false}, "last_changed": "2018-01-07T17:03:30.068831+00:00", "last_updated": "2018-01-07T17:03:30.068831+00:00"}, "new_state": {"entity_id": "sensor.cpu_used", "state": "20", "attributes": {"unit_of_measurement": "%", "friendly_name": "CPU used", "icon": "mdi:memory", "homebridge_hidden": true, "emulated_hue_hidden": true, "google_assistant": false}, "last_changed": "2018-01-07T17:04:00.821761+00:00", "last_updated": "2018-01-07T17:04:00.821761+00:00"}}, "origin": "LOCAL", "time_fired": "2018-01-07T17:04:00.821963+00:00"}
[2018-1-8 00:04:01] [HomeAssistant] Received event: {"event_type": "state_changed", "data": {"entity_id": "sensor.received_eth0", "old_state": {"entity_id": "sensor.received_eth0", "state": "3883.0", "attributes": {"unit_of_measurement": "MiB", "friendly_name": "Received eth0", "icon": "mdi:server-network", "homebridge_hidden": true, "emulated_hue_hidden": true, "google_assistant": false}, "last_changed": "2018-01-07T17:03:29.871687+00:00", "last_updated": "2018-01-07T17:03:29.871687+00:00"}, "new_state": {"entity_id": "sensor.received_eth0", "state": "3883.3", "attributes": {"unit_of_measurement": "MiB", "friendly_name": "Received eth0", "icon": "mdi:server-network", "homebridge_hidden": true, "emulated_hue_hidden": true, "google_assistant": false}, "last_changed": "2018-01-07T17:04:00.913559+00:00", "last_updated": "2018-01-07T17:04:00.913559+00:00"}}, "origin": "LOCAL", "time_fired": "2018-01-07T17:04:00.913747+00:00"}
[2018-1-8 00:04:01] [HomeAssistant] Received event: {"event_type": "state_changed", "data": {"entity_id": "sensor.ram_available", "old_state": {"entity_id": "sensor.ram_available", "state": "364.5", "attributes": {"unit_of_measurement": "MiB", "friendly_name": "RAM available", "icon": "mdi:memory", "homebridge_hidden": true, "emulated_hue_hidden": true, "google_assistant": false}, "last_changed": "2018-01-07T17:03:30.100077+00:00", "last_updated": "2018-01-07T17:03:30.100077+00:00"}, "new_state": {"entity_id": "sensor.ram_available", "state": "357.4", "attributes": {"unit_of_measurement": "MiB", "friendly_name": "RAM available", "icon": "mdi:memory", "homebridge_hidden": true, "emulated_hue_hidden": true, "google_assistant": false}, "last_changed": "2018-01-07T17:04:01.021277+00:00", "last_updated": "2018-01-07T17:04:01.021277+00:00"}}, "origin": "LOCAL", "time_fired": "2018-01-07T17:04:01.021472+00:00"}
[2018-1-8 00:04:01] [HomeAssistant] Received event: {"event_type": "state_changed", "data": {"entity_id": "sensor.sent_eth0", "old_state": {"entity_id": "sensor.sent_eth0", "state": "1372.2", "attributes": {"unit_of_measurement": "MiB", "friendly_name": "Sent eth0", "icon": "mdi:server-network", "homebridge_hidden": true, "emulated_hue_hidden": true, "google_assistant": false}, "last_changed": "2018-01-07T17:03:29.796558+00:00", "last_updated": "2018-01-07T17:03:29.796558+00:00"}, "new_state": {"entity_id": "sensor.sent_eth0", "state": "1372.3", "attributes": {"unit_of_measurement": "MiB", "friendly_name": "Sent eth0", "icon": "mdi:server-network", "homebridge_hidden": true, "emulated_hue_hidden": true, "google_assistant": false}, "last_changed": "2018-01-07T17:04:01.060019+00:00", "last_updated": "2018-01-07T17:04:01.060019+00:00"}}, "origin": "LOCAL", "time_fired": "2018-01-07T17:04:01.061081+00:00"}
[2018-1-8 00:04:01] [HomeAssistant] Received event: {"event_type": "state_changed", "data": {"entity_id": "sensor.ram_used", "old_state": {"entity_id": "sensor.ram_used", "state": "62.5", "attributes": {"unit_of_measurement": "%", "friendly_name": "RAM used", "icon": "mdi:memory", "homebridge_hidden": true, "emulated_hue_hidden": true, "google_assistant": false}, "last_changed": "2018-01-07T17:03:30.149204+00:00", "last_updated": "2018-01-07T17:03:30.149204+00:00"}, "new_state": {"entity_id": "sensor.ram_used", "state": "63.2", "attributes": {"unit_of_measurement": "%", "friendly_name": "RAM used", "icon": "mdi:memory", "homebridge_hidden": true, "emulated_hue_hidden": true, "google_assistant": false}, "last_changed": "2018-01-07T17:04:01.134439+00:00", "last_updated": "2018-01-07T17:04:01.134439+00:00"}}, "origin": "LOCAL", "time_fired": "2018-01-07T17:04:01.135013+00:00"}
```

**Comment by frenck** [comment]
Thank you @Gio76, I think there is missing something there, not sure.

Will look into it asap. Planning on updating this add-on this week.

**Comment by Gio76** [comment]
Thanks @frenck! looking forward to hearing from you. Hopefully the update fixes the issue I'm experiencing

**Comment by Gio76** [comment]
Hi @frenck, thanks for the 2.3.0 update, really appreciate the work you are putting into this addon. Unfortunately I still have the same _slow update_ issue. Every time I open the Home App, I have to wait roughly 7/8 seconds before I am able to view the status/interact with the HA accessory.

_**My current setup:**_
- Hassio 0.61.1
- Supervisor 0.83
- Homebridge addon 2.3.0

_**Homebridge addon Options:**_
```{
  "log_level": "info",
  "avahi_interfaces": "",
  "avahi_hostname": "",
  "avahi_domainname": "local",
  "enable_ipv6": true,
  "packages": [],
  "init_commands": [],
  "plugins": []
}
```

_**Homebridge config.json:**_
```{
  "bridge": {
    "name": "Home Assistant",
    "username": "B8:27:EB:A8:84:FA",
    "port": 51826,
    "pin": "087-70-688"
  },
  "description": "Homebridge for Home Assistant",
  "accessories": [],
  "platforms": [
    {
      "platform": "HomeAssistant",
      "name": "HomeAssistant",
      "host": "http://hassio/homeassistant",
      "password": "",
      "default_visibility": "visible",
      "supported_types": [
        "automation",
        "binary_sensor",
        "climate",
        "cover",
        "device_tracker",
        "fan",
        "group",
        "light",
        "lock",
        "media_player",
        "remote",
        "scene",
        "script",
        "sensor",
        "switch"
      ],
      "logging": true,
      "verify_ssl": true
    }
  ]
}
```

**Comment by frenck** [comment]
Sorry, this version does not contain any changes regarding this GitHub issue.
Please read the CHANGELOG before making assumptions.

**Comment by Gio76** [comment]
My apologies @frenck, in your previous message you mentioned that you were planning an update so I assumed, since you said you were going to look into this Gitgub issue, that it the 2.3.0 update was going to address my problem as well, my bad!

> Will look into it asap. Planning on updating this add-on this week.

**Comment by frenck** [comment]
This is an opensource project which is something done by volunteers.
I'm currently in the process of fixing another bug affecting ALL my add-ons, which has a higher priority (hence this latest update).

I don't think I can fix this anytime soon. I'll be away for a month next week.

**Comment by bl8rnr** [comment]
@Gio76, I used to run my home-assistant on an old mac mini running linux but recently changed to almost the same setup as yours.

Before the change I had no problems with the homebridge plugin but after the change I experienced exactly the same issue as you described above. 

My config was the recommended host `http://hassio/homeassistant` with no password and `verify_ssl` was `true`.

I changed the host `https://myhostname.duckdns.org:8123` with my password and now everything is working as before. 

It is not clear from your problem description above if you have a valid let's encrypt certificate installed, but that is the only difference I can think of.

Hope this helps.
